lib/preset-matcher.js

```diff
@@ -0,0 +1,82 @@
+module.exports = function presetMatcher (presets) {
+  // Index of presets by (geometry, tag key).
+  var index = {
+    point: {},
+    vertex: {},
+    line: {},
+    area: {},
+    relation: {}
+  }
+
+  for (var name in presets) {
+    var preset = Object.assign({}, presets[name], {
+      id: name
+    })
+    var geometry = preset.geometry || []
+
+    for (var j = 0; j < geometry.length; j++) {
+      var g = index[geometry[j]]
+      for (var k in preset.tags) {
+        k = k.replace(':', '_')
+        ;(g[k] = g[k] || []).push(preset)
+      }
+    }
+  }
+
+  return match
+
+  function match (feature) {
```

Comment:
Nit: if you made `index` a parameter, you could break this function out of the closure and make it pure.